apps/core/src/utils/transaction/helpers/getObjectChangeSummary.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import {
+    COIN_TYPE_ARG_REGEX,
+    type SuiTransactionBlockResponse,
+    type SuiAddress,
+    type DryRunTransactionBlockResponse,
+} from '@mysten/sui.js';
+
+export const getObjectChangeSummary = (
+    transaction: DryRunTransactionBlockResponse | SuiTransactionBlockResponse,
+    currentAddress?: SuiAddress | null
+) => {
+    const { objectChanges } = transaction;
+    if (!objectChanges) return null;
+
+    const mutated = objectChanges.filter((change) => change.type === 'mutated');
+
+    const created = objectChanges
+        .filter(
+            (change) =>
+                change.type === 'created' && change.sender === currentAddress
+        )
+        .map((change) => ({
+            ...change,
+            minted:
+                change.type === 'created' &&
+                change.sender === currentAddress &&
+                typeof change.owner === 'object' &&
+                'AddressOwner' in change.owner &&
+                change.owner.AddressOwner === currentAddress &&
+                !change.objectType.match(COIN_TYPE_ARG_REGEX),
+        }));
+
+    const transferred = objectChanges.filter(
```

Comment:
I think you also _could_ use the transaction effects to compute this more efficiently.